worker/src/RTC/RtpPacket.cpp

```diff
@@ -600,12 +598,105 @@ namespace RTC
 
 			auto currentLen = extension->len + 1;
 
-			// Fill with 0's if new length is minor.
-			if (len < currentLen)
-				std::memset(extension->value + len, 0, currentLen - len);
+			MS_DEBUG_DEV(
+			  "set extension id: %" PRIu8 ", length:%" PRIu8 ", currentLen:%" PRIu8 " value:%s",
+			  id,
+			  len,
+			  currentLen,
+			  value.c_str());
+
+			// If len == currentLen, just memcpy.
+			if (len == currentLen)
+			{
+				std::memcpy(extension->value, value.c_str(), len);
+			}
+			// If len != currentLen, means we may need to shift extension or payload.
+			else
+			{
+				uint8_t* extensionStart    = reinterpret_cast<uint8_t*>(this->headerExtension) + 4;
+				uint8_t* extensionEnd      = extensionStart + GetHeaderExtensionLength();
+				uint8_t* ptr               = extensionStart;
+				size_t extensionsTotalSize = static_cast<size_t>(len + 1);
+				uint8_t* ptr1              = ptr;
```

Comment:
Please, rename to `lastExtension`, since it points exactly to the last extension that has been evaluated, while `ptr` keeps moving forward among the 0s.